cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -438,22 +438,44 @@ func extractExtendedResourcesFromKubeEnv(kubeEnv KubeEnv) (apiv1.ResourceList, e
 		klog.Warningf("error while obtaining extended_resources from AUTOSCALER_ENV_VARS; %v", err)
 		return nil, err
 	}
-
-	if !found {
-		return apiv1.ResourceList{}, nil
+	var extendedResourcesMap map[string]string
+	if found {
+		extendedResourcesMap, err = parseKeyValueListToMap(extendedResourcesAsString)
+		if err != nil {
+			return apiv1.ResourceList{}, err
+		}
+	} else {
+		extendedResourcesMap = make(map[string]string)
 	}
-
-	extendedResourcesMap, err := parseKeyValueListToMap(extendedResourcesAsString)
+	nodeLabelsAsString, found, err := extractAutoscalerVarFromKubeEnv(kubeEnv, "node_labels")
 	if err != nil {
-		return apiv1.ResourceList{}, err
+		klog.Warningf("error while obtaining node_labels from AUTOSCALER_ENV_VARS; %v", err)
+		return nil, err
+	}
+	if found {
+		nodeLabelsMap, err := parseKeyValueListToMap(nodeLabelsAsString)
+		if err != nil {
+			return apiv1.ResourceList{}, err
+		}
+		const extendedResourcesKeyPrefix = "clusterautoscaler-nodetemplate-resources-"
```

Comment:
The Amazon and AWS providers (where these additional configuration options and flags are much better documented…) use `k8s.io/cluster-autoscaler/node-template/…` as the prefix for what the users should/could self-configure, maybe it would be wise to adhere to this pattern here if possible.

Reply:
No it is not possible. You can have only one `\` at most in node node labels. AWS has a separation between node labels and autoscaling group templates, GKE does not have such separation, and templates just copy attributes from node labels.